ExampleMod/Content/NPCs/MinionBoss/MinionBossBody.cs

```diff
@@ -0,0 +1,473 @@
+			//(Lesser Healing Potion). If you wanted to change it, simply write "potionType = ItemID.HealingPotion;" or any other potion type
+		}
+
+		public override void FindFrame(int frameHeight) {
+			//This NPC animates with a simple "go from start frame to final frame, and loop back to start frame" rule
+			//In this case: First stage: 0-1-2-0-1-2, Second stage: 3-4-5-3-4-5, 5 being "total frame count - 1"
+			int startFrame = 0;
+			int finalFrame = 2;
+
+			if (SecondStage) {
+				startFrame = 3;
+				finalFrame = Main.npcFrameCount[NPC.type] - 1;
+
+				if (NPC.frame.Y < startFrame * frameHeight) {
+					//If we were animating the first stage frames and then switch to second stage, immediately change to the start frame of the second stage
+					NPC.frame.Y = startFrame * frameHeight;
+				}
+			}
+
+			int frameSpeed = 5;
+			NPC.frameCounter += 0.5f;
+			NPC.frameCounter += NPC.velocity.Length() / 10f; //Make the counter go faster with more movement speed
+			if (NPC.frameCounter > frameSpeed) {
+				NPC.frameCounter = 0;
+				NPC.frame.Y += frameHeight;
+
+				if (NPC.frame.Y > finalFrame * frameHeight) {
+					NPC.frame.Y = startFrame * frameHeight;
+				}
+			}
+		}
+
+		public override void HitEffect(int hitDirection, double damage) {
+			//If the NPC dies, spawn gore and play a sound
+			if (NPC.life <= 0) {
+				//These gores work by simply existing as a texture inside the ExampleMod/Gores folder. They won't be recognized in any other folder
+				int backGoreType = ModContent.Find<ModGore>("ExampleMod/MinionBossBody_Back").Type;
+				int frontGoreType = ModContent.Find<ModGore>("ExampleMod/MinionBossBody_Front").Type;
+
+				for (int i = 0; i < 2; i++) {
+					Gore.NewGore(NPC.position, new Vector2(Main.rand.Next(-6, 7), Main.rand.Next(-6, 7)), backGoreType);
+					Gore.NewGore(NPC.position, new Vector2(Main.rand.Next(-6, 7), Main.rand.Next(-6, 7)), frontGoreType);
+				}
+
+				SoundEngine.PlaySound(SoundID.Roar, NPC.Center, 0);
+			}
+		}
+
+		public override void AI() {
+			//This should almost always be the first code in AI() as it is responsible for finding the proper player target.
+			if (NPC.target < 0 || NPC.target == 255 || Main.player[NPC.target].dead || !Main.player[NPC.target].active) {
+				NPC.TargetClosest();
+			}
+
+			Player player = Main.player[NPC.target];
+
+			if (player.dead) {
+				//If the targeted player is dead, flee
+				NPC.velocity.Y -= 0.04f;
+				//This method makes it so when the boss is in "despawn range" (outside of the screen), it despawns in 10 ticks
+				NPC.EncourageDespawn(10);
+				return;
+			}
+
+			SpawnMinions();
+
+			CheckSecondStage();
+
+			//Be invulnerable during the first stage
+			NPC.dontTakeDamage = !SecondStage;
+
+			if (SecondStage) {
+				DoSecondStage(player);
+			}
+			else {
+				DoFirstStage(player);
+			}
+		}
+
+		private void SpawnMinions() {
+			if (!SpawnedMinions) {
+				SpawnedMinions = true;
+
+				if (Main.netMode != NetmodeID.MultiplayerClient) {
```

Comment:
Short-circuit to reduce indentation more frequently.

Reply:
got rid of two indentations